content/navigation.tsx

```diff
@@ -3,6 +3,9 @@ import React from 'react';
 import ProductIcon from '../src/components/ProductIcon';
 import { IItem } from '../src/types/Item';
 
+const now = new Date();
```

Comment:
This and the firebase redirect should be hardcoded.  The reason being, at the beginning of the year the page won't exist until a new blog post is created, eg On Jan 1 2023, it's very likely that the most recent post is from Dec 2022, so no 2023 page will exist